### PR TITLE
Deduplicate slashes for sigv4 signature

### PR DESCRIPTION
**Pull request by importhuman**
Signed-off-by: Ujjwal Goyal <importujjwal@gmail.com>

For issue [10200 on the prometheus repo](https://github.com/prometheus/prometheus/issues/10200).
Not sure if this is all that is required.

### CODE REVIEW

**Comment by roidelapluie** [comment]
Can we have a test?

File: sigv4/sigv4.go [hunk 1]
```diff
@@ -115,6 +115,9 @@ func (rt *sigV4RoundTripper) RoundTrip(req *http.Request) (*http.Response, error
 	}()
 	req.Body = ioutil.NopCloser(seeker)
 
+	// Escape double slashes if present
```
Comment (roidelapluie) [inline sigv4/sigv4.go:118]:
```suggestion
	// Escape URL like documented in <xxx>.
```

**Comment by importhuman** [comment]
Did you mean a test like this?

It's failing currently, because it's not being escaped. `req.URL.EscapedPath()` uses the `RawPath` of the URL, which is coming out to be `""`, in which case it doesn't escape. Any suggestions?

**Review by SuperQ — APPROVED** [review]
LGTM

**Comment by importhuman** [comment]
Were the tests not run in CI, because I expected the test I added to fail :thinking:
Don't think this is ready to merge.

**Comment by SuperQ** [comment]
Yes, we run a `make test` on this package in CI.

**Comment by SuperQ** [comment]
Looks like the `make test` incorrectly skips `go test`. 

Fixed in https://github.com/prometheus/common/pull/365.

**Review by kakkoyun — APPROVED** [review]

**Comment by importhuman** [comment]
Ah, it's inside circleci, didn't realize that.

Screenshot from 2022-03-28 17-24-51
This is what I'm concerned about. Why is it passing on the CI?

**Comment by SuperQ** [comment]
Ok, a rebase here will make the tests run correctly.

**Comment by roidelapluie** [comment]
It turns out that the underlying API is already calling EscapePath. The fix for the user's issue is to use path.Clean. I have updated the pull request to reflect this and added a test.

WDYT @importhuman ?

**Comment by importhuman** [comment]
LGTM @roidelapluie

**Comment by roidelapluie** [comment]
Thanks!!!

**Pull request merged**